sql/core/src/main/scala/org/apache/spark/sql/execution/Expand.scala

```diff
@@ -41,14 +41,34 @@ case class Expand(
   // as UNKNOWN partitioning
   override def outputPartitioning: Partitioning = UnknownPartitioning(0)
 
+  override def outputsUnsafeRows: Boolean = child.outputsUnsafeRows
+  override def canProcessUnsafeRows: Boolean = true
+  override def canProcessSafeRows: Boolean = true
+
+  protected val cleanedProjections = {
+    if (outputsUnsafeRows) {
+      projections.map { exprs =>
+        exprs.map(_ transform {
+          case CreateStruct(children) => CreateStructUnsafe(children)
```

Comment:
Should we do this inside of `UnsafeProjection.create`?  It seems weird to have to remember to do it in each place we use it.